Dfe.Data.SearchPrototype/Web/Tests/PageIntegrationTests/SearchResultsPageTests.cs

```diff
@@ -28,13 +31,21 @@ public HomePageTests(PageWebApplicationFactory<Program> factory, ITestOutputHelp
         }
 
         [Fact]
-        public async Task Search_Title_IsDisplayed()
-        {
-            var response = await _client.GetAsync(uri);
-
-            var document = await HtmlHelpers.GetDocumentAsync(response);
-
-            document.GetElementText(HomePage.Heading.Criteria).Should().Be("Search prototype");
+        public async Task Search_Title_Displayed()
+        {
+            // Act
+            Response response = await GetResponseAsync("/");
+
+            // Assert
+            new HomePage(response.DomQueryClient).GetHeading().Should().Be("Search prototype");

```

Comment:
`DomQueryClient` would likely be a scoped dependency per each test, because the DOM it operates on, will be provided at runtime.